core/src/test/scala/unit/kafka/server/CreateTopicsRequestTest.scala

```diff
@@ -0,0 +1,232 @@
+        assertEquals("The topic should have the correct replication factor", replication, metadataForTopic.partitionMetadata.asScala.head.replicas.size)
+      }
+
+      // Verify controller broker has the correct metadata
+      verifyMetadata(controllerSocketServer)
+      // Wait until metadata is propagated and validate non-controller broker has the correct metadata
+      TestUtils.waitUntilMetadataIsPropagated(servers, topic, 0)
+      verifyMetadata(notControllerSocketServer)
+    }
+  }
+
+  @Test
+  def testErrorCreateTopicsRequests() {
+    val timeout = 10000
+    val existingTopic = "existing-topic"
+    TestUtils.createTopic(zkUtils, existingTopic, 1, 1, servers)
+
+    // Basic
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map(existingTopic -> new CreateTopicsRequest.TopicDetails(1, 1.toShort)).asJava, timeout),
+      Map(existingTopic -> Errors.TOPIC_ALREADY_EXISTS))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-partitions" -> new CreateTopicsRequest.TopicDetails(-1, 1.toShort)).asJava, timeout),
+      Map("error-partitions" -> Errors.INVALID_PARTITIONS))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-replication" -> new CreateTopicsRequest.TopicDetails(1, (numBrokers + 1).toShort)).asJava, timeout),
+      Map("error-replication" -> Errors.INVALID_REPLICATION_FACTOR))
+    val invalidConfig = Map("not.a.property" -> "error").asJava
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-config" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort, invalidConfig)).asJava, timeout),
+      Map("error-config" -> Errors.INVALID_CONFIG))
+    val invalidAssignments = replicaAssignmentToJava(Map(0 -> List(0, 1), 1 -> List(0)))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-assignment" -> new CreateTopicsRequest.TopicDetails(invalidAssignments)).asJava, timeout),
+      Map("error-assignment" -> Errors.INVALID_REPLICA_ASSIGNMENT))
+
+    // Partial
+    validateErrorCreateTopicsRequests(
+      new CreateTopicsRequest(Map(
+        existingTopic -> new CreateTopicsRequest.TopicDetails(1, 1.toShort),
+        "partial-partitions" -> new CreateTopicsRequest.TopicDetails(-1, 1.toShort),
+        "partial-replication" -> new CreateTopicsRequest.TopicDetails(1, (numBrokers + 1).toShort),
+        "partial-assignment" -> new CreateTopicsRequest.TopicDetails(invalidAssignments),
+        "partial-none" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort)).asJava, timeout),
+      Map(
+        existingTopic -> Errors.TOPIC_ALREADY_EXISTS,
+        "partial-partitions" -> Errors.INVALID_PARTITIONS,
+        "partial-replication" -> Errors.INVALID_REPLICATION_FACTOR,
+        "partial-assignment" -> Errors.INVALID_REPLICA_ASSIGNMENT,
+        "partial-none" -> Errors.NONE
+      )
+    )
+    validateTopicExists("partial-none")
+
+    // Timeout
```

Comment:
Could we add a comment saying that we expect the request never be completed within 1ms?